### ISSUE TITLE
Asmjs compilation error on assigning float/signed/unsigned return values of simd builtin to a double var

### ISSUE THREAD

**Issue by arunetm** [Microsoft/ChakraCore#933]
Simd builtin return values should be allowed to be coerced to all the types where a valid coercion exists  as allowed by the asm.js spec.
http://asmjs.org/spec/latest/#unary-operators

**Repro Case**
```javascript
function asmModule(stdlib, imports, buffer) {
    "use asm";
    var fround = stdlib.Math.fround;
   var f4 = stdlib.SIMD.Float32x4; 
    var f4ext = f4.extractLane;
    function func1()
    {
        //var retn = fround(0);
        var retn = 0.0;
        var st = f4(1.0, 2.0, 3.0, 4.0);
        retn = +f4ext(st,0);
    }    
    return {func1:func1};
}
```

**Expected Results**
Asm.js Compilation completed successfully

**Actual Result**
Asm.js Compilation Error function : asmModule::func1

**Comment by arunetm** [comment]
@dilijev This bug needs to be fixed for RS1. Can you please mark this issue as RS1?

**Comment by arunetm** [comment]
@dilijev please close this issue. the fix is merged to Release/1.2

**Comment by dilijev** [comment]
@arunetm The issue will be closed automatically when this change is merged into master (soon)
